subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/util/Optional.java

```diff
@@ -0,0 +1,37 @@
+package org.shipkit.internal.gradle.util;
+
+import org.shipkit.internal.util.ArgumentValidation;
+
+import java.util.NoSuchElementException;
+
+/**
+ * We need to migrate to Java8 and remove this.
+ */
+public class Optional<T> {
+
+    private final T value;
+
+    private Optional(T value) {
+        this.value = value;
+    }
+
+    public T get() {
+        if (value == null) {
+            throw new NoSuchElementException("No value present");
+        }
+        return value;
+    }
+
+    public static <T> Optional<T> of(T object) {
+        ArgumentValidation.notNull();
```

Comment:
Let's pass the `object` and a message to `ArgumentValidation.notNull` and migrate `Optional.of(null)` usage in `UpgradeDependencyPluginTest` to `Optional.ofNullable(null)`.
Afterwards we are ready to merge.

Reply:
haha, I love code reviews. Thank you for catching this dumb mistake. Imagine that I decided not to write unit tests around it because I deemed it too simple...